src/main/kotlin/ink/abb/pogo/scraper/controllers/BotController.kt

```diff
@@ -56,4 +70,160 @@ class BotController {
     fun stopBot(@PathVariable name: String): Boolean {
         return service.doWithBot(name) { it.stop() }
     }
+    
+    
+    @RequestMapping("/bot/{name}/pokemons")
+    fun listPokemons(@PathVariable name: String): List<PokemonData> {
+      
+        val data = service.getBotContext(name).api.inventories.pokebank.pokemons
+        val returnData = mutableListOf<PokemonData>()
+        for(pokemon in data) {
+            returnData.add(PokemonData().buildFromPokemon(pokemon))
+        }
+
+        return returnData
+    }
+
+    @RequestMapping("/bot/{name}/pokemon/{id}/{method}")
```

Comment:
Yes I know, I made this to test more practically. Your suggestion or a different end point for each action